front/pages/api/lookup.ts

```diff
@@ -0,0 +1,174 @@
+
+const ExternalUserCodec = t.type({
+  email: t.string,
+  email_verified: t.boolean,
+  sub: t.string,
+});
+
+type LookupResponseBody = UserLookupResponse | WorkspaceLookupResponse;
+
+const UserLookupSchema = t.type({
+  user: ExternalUserCodec,
+});
+
+const WorkspaceLookupSchema = t.type({
+  workspace: t.string,
+});
+
+const ResourceType = t.union([t.literal("user"), t.literal("workspace")]);
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorResponse<LookupResponseBody>>
+): Promise<void> {
+  if (req.method !== "POST") {
+    return apiError(req, res, {
+      status_code: 405,
+      api_error: {
+        type: "method_not_supported_error",
+        message: "Only POST requests are supported",
+      },
+    });
+  }
+
+  const bearerTokenRes = await getBearerToken(req);
+  if (bearerTokenRes.isErr()) {
+    return apiError(req, res, {
+      status_code: 401,
+      api_error: {
+        type: "not_authenticated",
+        message: "The request does not have valid authentication credentials",
+      },
+    });
+  }
+
+  if (bearerTokenRes.value !== config.getRegionResolverSecret()) {
+    return apiError(req, res, {
+      status_code: 401,
+      api_error: {
+        type: "invalid_basic_authorization_error",
+        message: "Invalid token",
+      },
+    });
+  }
+  const { resource } = req.query;
+  const resourceValidation = ResourceType.decode(resource);
+  if (isLeft(resourceValidation)) {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Invalid resource type. Must be 'user' or 'workspace'",
+      },
+    });
+  }
+
+  let response: LookupResponseBody | null = null;
+  switch (resourceValidation.right) {
+    case "user":
+      {
+        const bodyValidation = UserLookupSchema.decode(req.body);
+        if (isLeft(bodyValidation)) {
+          const pathError = reporter.formatValidationErrors(
+            bodyValidation.left
+          );
+          return apiError(req, res, {
+            status_code: 400,
+            api_error: {
+              type: "invalid_request_error",
+              message: `Invalid request body for user lookup: ${pathError}`,
+            },
+          });
+        }
+        response = await handleLookupUser(bodyValidation.right);
+      }
+      break;
+    case "workspace": {
+      const bodyValidation = WorkspaceLookupSchema.decode(req.body);
+      if (isLeft(bodyValidation)) {
+        const pathError = reporter.formatValidationErrors(bodyValidation.left);
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message: `Invalid request body for user lookup ${pathError}`,
+          },
+        });
+      }
+      response = await handleLookupWorkspace(bodyValidation.right);
+    }
+  }
+  res.status(200).json(response);
+  return;
+}
+
+async function handleLookupUser(
```

Comment:
This should return a list of object for each workspace relationship

Here it's weird as you return a workspace object but that can have id (should be sId) undefined that will have sId undefined if the user is fully signed up in an existing worksapce

Reply:
After second thought with @Fraggle we actually don't need workspace information. We just want to know whether the user exists somewhere in the DB

Reply:
Will simply return:
```
{ "status": "unknown" | "existing"}
```